### PR TITLE
Add clip conv layer

### PR DESCRIPTION
**Pull request by DanialTaheri**
# What does this PR do ?

Replace the linear layer in the VIT backbone clip with the convolution layer.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Changes in both vit_backbone.py and convert_external_clip_to_nemo.py

# Usage
More similarity with the existing Clip models in the literature.

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [X] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
Attaching the successful run of the [convert_external_clip_to_nemo.py]
Screenshot 2024-04-08 at 1 04 33 PM

### CODE REVIEW

File: examples/multimodal/vision_language_foundation/clip/convert_external_clip_to_nemo.py [hunk 1]
```diff
@@ -55,8 +55,8 @@
 
 def get_args():
     parser = ArgumentParser()
-    parser.add_argument("--arch", type=str, default="ViT-H-14")
-    parser.add_argument("--version", type=str, default="laion2b_s32b_b79k")
+    parser.add_argument("--arch", type=str, default="huggingface")
+    parser.add_argument("--version", type=str, default="openai/clip-vit-base-patch32")
```
Comment (yaoyu-33) [inline examples/multimodal/vision_language_foundation/clip/convert_external_clip_to_nemo.py:59]: note for hugging face, you should set version to huggingface and set arch to model name, check line 255 below

Comment (DanialTaheri) [reply examples/multimodal/vision_language_foundation/clip/convert_external_clip_to_nemo.py]: Thanks, @yaoyu-33, for the comment. I resolved the typo.

Comment (DanialTaheri) [reply examples/multimodal/vision_language_foundation/clip/convert_external_clip_to_nemo.py]: Attached a picture that shows the successful run of [convert_external_clip_to_nemo.py]

**Comment by Victor49152** [comment]
I have verified the conversion works and this new converted ckpt works with SD. Thanks!!

**Comment by yaoyu-33** [comment]
jenkins

**Review by yaoyu-33 — APPROVED** [review]

**Pull request merged**